File: app/src/main/java/lab/inconcept/musiclibrary/helper/EmailValidateHelper.java
package lab.inconcept.musiclibrary.helper;

import java.util.regex.Pattern;

public class EmailValidateHelper {
    public static boolean isValid(String email) {
        final String emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\." +
                "[a-zA-Z0-9_+&*-]+)*@" +
                "(?:[a-zA-Z0-9-]+\\.)+[a-z" +
                "A-Z]{2,7}$";

        final Pattern pattern = Pattern.compile(emailRegex);
        if (email == null) {
            return false;
        }
        return pattern.matcher(email).matches();
    }
}

File: app/src/main/java/lab/inconcept/musiclibrary/helper/EmailValidateHelper.kt
package lab.inconcept.musiclibrary.helper

import java.util.regex.Pattern

object EmailValidateHelper {
    fun isValid(email: String?): Boolean {
        val emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\." +
                "[a-zA-Z0-9_+&*-]+)*@" +
                "(?:[a-zA-Z0-9-]+\\.)+[a-z" +
                "A-Z]{2,7}$"
        val pattern = Pattern.compile(emailRegex)
        return if (email == null) {
            false
        } else pattern.matcher(email).matches()
    }
}